### ISSUE TITLE
[Bug]: failed to apply snapshot, aoe: db not found

### ISSUE THREAD

**Issue by lni** [matrixorigin/matrixone#1354]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 77f928d5f686b88b5bde0144ea49d5d3ab99ddfb from https://github.com/jiangxinmeng1/matrixone-1
- Hardware parameters:
- OS type: MacOS 12.0.1
- Others:
```

### Actual Behavior

Got the following error when applying snapshot into aoe. 

```
{"level":"error","ts":"2021-12-12 12:22:18.184","logger":"cube.store","caller":"raftstore/replica_snapshot.go:100","msg":"failed to recover from the snapshot","case":"TestSnapshot","store":"node-2","store-id":35,"shard-id":121,"replica-id":124,"snapshot-index":2,"error":"aoe: db not found"}
panic: aoe: db not found
```

### Expected Behavior

No error when applying snapshot

### Steps to Reproduce

```markdown
Run TestSnapshot using the above mentioned version.
```

### Additional information

data folder from the test 

[test.tar.gz](https://github.com/matrixorigin/matrixone/files/7698104/test.tar.gz)

**Comment by XuPeng-SH** [comment]
The current version of the snapshot implementation is a **Replace** semantics (replacing a replica), without considering the **Install** semantics (install as a new replica). #1358 will fill in the missing logic.

**Comment by XuPeng-SH** [comment]
should be fixed by #1358
